src/utils/wrap/mpi/dpar_mpi.c

```diff
@@ -348,3 +353,23 @@ par_comm_free(uint32_t pcom)
 
 	return 0;
 }
+
+bool
+par_version_compatible(uint32_t version)
+{
+	uint32_t major = version >> DPAR_VERSION_SHIFT;
+	uint32_t minor = version & DPAR_VERSION_MASK;
+
+	/** Major version difference means this library is incompatible with the caller */
+	if (major != DPAR_MAJOR)
+		return false;
+
+	/** For minor version, it is only incompatible if the caller minor version is greater
+	 *  than the minor version of the library.  This means that new APIs are expected
+	 *  but are not defined.
+	 */
+	if (minor > DPAR_MINOR)
```

Comment:
Here the DPAR_MINOR is zero, so any non-zero minor will be incompatible?

Reply:
> > It's only incompatible if the user is trying to use a library that doesn't have an API they need, not the other way around.
> > Required-githooks: true
> > Signed-off-by: Jeff Olivier [jeffrey.v.olivier@intel.com](mailto:jeffrey.v.olivier@intel.com)
> 
> Currently, there is only one caller for par_version_compatible(). That is under the case of the expected API names have been successfully loaded, means that the library has the need API. According this commit comment, it should be counted as 'compatible', that is confused.

Ah, I see, you are right.  Let me fix that.  This is trying to fix a coverity issue because it doesn't the version check.